tests/pass-dep/tokio/sleep.rs

```diff
@@ -1,14 +1,17 @@
-//@compile-flags: -Zmiri-disable-isolation -Zmiri-permissive-provenance -Zmiri-backtrace=full
+//@compile-flags: -Zmiri-isolation-error=warn -Zmiri-permissive-provenance -Zmiri-backtrace=full
 //@only-target-x86_64-unknown-linux: support for tokio only on linux and x86
 
 use tokio::time::{sleep, Duration, Instant};
 
 #[tokio::main]
 async fn main() {
+    // We are using a very short time for sleep, 1 millisecond, (and ensure sleep waits for at
+    // least 1000 microseconds, the same as 1 millisecond) because this test is using miri's
+    // virtual clock due to isolation. The virtual clock works by using 1 code block passed as 10
+    // nanoseconds. While writing this test, I saw 1,000,000 basic blocks had been executed pretty
+    // quickly which would be the equivalent of 1 millisecond.
```

Comment:
@pvdrz I think we may want to look into increasing the tick size in isolation mode. A second of isolation mode sleep time should probably be not more than 10s real time on a regular desktop CPU.

Reply:
are you telling me that you want `miri` to be slower? :trollface:

Reply:
So instead of 10ns we should make it 100ns? Sure, feel free to make a PR.

Reply:
The thing is that if you sleep for 1s without isolation, the test runs in user-visible 1s (or close enough). If you sleep for 1s in isolation, it takes minutes. I think isolation should at least be in the right ballpark.

> So instead of 10ns we should make it 100ns? Sure, feel free to make a PR.

more like 1000ns, but yes

Reply:
> The thing is that if you sleep for 1s without isolation, the test runs in user-visible 1s (or close enough). If you sleep for 1s in isolation, it takes minutes. I think isolation should at least be in the right ballpark.

Oh, but something here doesn't make sense. With isolation, it should always take the same amount of real time, no matter how long you sleep. Sleeping 60s should still only take a few ms of real time at most. And for sure changing the clock timer should not change how much real time a program takes. Something else is wrong here and this needs to be investigated.

Reply:
This was resolved in https://github.com/rust-lang/miri/pull/2816